File: jars/src/org/niatahl/tahlan/shipsystems/InertialessDriveStats.java
package org.niatahl.tahlan.shipsystems;

import com.fs.starfarer.api.combat.MutableShipStatsAPI;
import com.fs.starfarer.api.combat.ShipAPI;
import com.fs.starfarer.api.impl.combat.BaseShipSystemScript;

import java.awt.*;

public class InertialessDriveStats extends BaseShipSystemScript {

    private static final Color SHIMMER_COLOR = new Color(119, 226, 176, 117);

    public void apply(MutableShipStatsAPI stats, String id, State state, float effectLevel) {

        ShipAPI ship = null;
        if (stats.getEntity() instanceof ShipAPI) {
            ship = (ShipAPI) stats.getEntity();
        } else {
            return;
        }

        if (state == State.OUT) {
            stats.getMaxSpeed().unmodify(id);
            stats.getMaxTurnRate().unmodify(id);
        } else {

            stats.getMaxSpeed().modifyFlat(id, 100f * effectLevel);
            stats.getAcceleration().modifyFlat(id, 1000f * effectLevel);
            stats.getDeceleration().modifyFlat(id, 1000f * effectLevel);
            stats.getTurnAcceleration().modifyFlat(id, 300f * effectLevel);
            stats.getMaxTurnRate().modifyMult(id, 1f + 2f * effectLevel);
        }

        ship.setJitterUnder(ship, SHIMMER_COLOR, 1f, 6, 2f, 6f);



    }

    public void unapply(MutableShipStatsAPI stats, String id) {
        stats.getMaxSpeed().unmodify(id);
        stats.getMaxTurnRate().unmodify(id);
        stats.getTurnAcceleration().unmodify(id);
        stats.getAcceleration().unmodify(id);
        stats.getDeceleration().unmodify(id);
    }

    public StatusData getStatusData(int index, State state, float effectLevel) {
        if (index == 0) {
            return new StatusData("improved maneuverability", false);
        }
        return null;
    }

}

File: jars/src/org/niatahl/tahlan/shipsystems/InertialessDriveStats.kt
package org.niatahl.tahlan.shipsystems

import com.fs.starfarer.api.combat.MutableShipStatsAPI
import com.fs.starfarer.api.combat.ShipAPI
import com.fs.starfarer.api.impl.combat.BaseShipSystemScript
import com.fs.starfarer.api.plugins.ShipSystemStatsScript
import com.fs.starfarer.api.plugins.ShipSystemStatsScript.StatusData
import java.awt.Color

class InertialessDriveStats : BaseShipSystemScript() {
    override fun apply(stats: MutableShipStatsAPI, id: String, state: ShipSystemStatsScript.State, effectLevel: Float) {
        val ship: ShipAPI = if (stats.entity is ShipAPI) {
            stats.entity as ShipAPI
        } else {
            return
        }
        if (state == ShipSystemStatsScript.State.OUT) {
            stats.maxSpeed.unmodify(id)
            stats.maxTurnRate.unmodify(id)
        } else {
            stats.maxSpeed.modifyFlat(id, 100f * effectLevel)
            stats.acceleration.modifyFlat(id, 1000f * effectLevel)
            stats.deceleration.modifyFlat(id, 1000f * effectLevel)
            stats.turnAcceleration.modifyFlat(id, 300f * effectLevel)
            stats.maxTurnRate.modifyMult(id, 1f + 2f * effectLevel)
        }
        ship.setJitterUnder(ship, SHIMMER_COLOR, 1f, 6, 2f, 4f)
    }

    override fun unapply(stats: MutableShipStatsAPI, id: String) {
        stats.maxSpeed.unmodify(id)
        stats.maxTurnRate.unmodify(id)
        stats.turnAcceleration.unmodify(id)
        stats.acceleration.unmodify(id)
        stats.deceleration.unmodify(id)
    }

    override fun getStatusData(index: Int, state: ShipSystemStatsScript.State, effectLevel: Float): StatusData? {
        return if (index == 0) {
            StatusData("improved maneuverability", false)
        } else null
    }

    companion object {
        private val SHIMMER_COLOR = Color(96, 251, 171, 117)
    }
}